src/nvim/terminal.c

```diff
@@ -246,11 +256,6 @@ Terminal *terminal_open(TerminalOptions opts)
   // Apply TermOpen autocmds _before_ configuring the scrollback buffer.
   apply_autocmds(EVENT_TERMOPEN, NULL, NULL, false, curbuf);
 
-  // Configure the scrollback buffer.
-  rv->sb_size = curbuf->b_p_scbk < 0
-                ? SB_MAX : (size_t)MAX(1, curbuf->b_p_scbk);
-  rv->sb_buffer = xmalloc(sizeof(ScrollbackLine *) * rv->sb_size);
```

Comment:
Haven't seen any failures, but I'll move it back to after the autocommand to be on the safe side.